contracts/TokenClaim.sol

```diff
@@ -0,0 +1,65 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "./interfaces/ITokenClaim.sol";
+import "./VotesToken.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+contract TokenClaim is FactoryFriendly, ITokenClaim {
+    using SafeERC20 for IERC20;
+
+    address public childToken;
+    address public parentToken;
+    uint256 public snapShotId;
+    uint256 public parentAllocation;
+    mapping(address => bool) claimed;
```

Comment:
Ah yes good catch, just pushed commit to make `claimed` be public.

Reply:
What do you think about changing the logic in `getClaimAmount()` to simply

```
return
    (VotesToken(parentToken).balanceOfAt(claimer, snapShotId) * parentAllocation) /
            VotesToken(parentToken).totalSupplyAt(snapShotId)
```

leave `claimed` public like it now is,

and updating the logic in `claimToken()` to check the `claimed` state, instead of checking if the amount to claim is 0. 

```
    function claimToken(address claimer) external {
        if (claimed[claimer]) revert AlreadyClaimed();

        uint256 amount = getClaimAmount(claimer); // Get user balance

        claimed[claimer] = true;

        IERC20(childToken).safeTransfer(claimer, amount); // transfer user balance

        emit TokenClaimed(parentToken, childToken, claimer, amount);
    }
```

Reason for my suggestion is that i just subjectively think it's cleaner, lol. I think it's cleaner that `getClaimAmount` returns the same value over time. This makes it more straightforward for clients to derive which of those two situations in my original comment is happening.

Reply:
hmmm but in both cases a client would have to both check the `getClaimAmount`, and then go back and look for a `TokenClaimed` event right?

What if we replace `getClaimAmount` with two getter functions. One being `getClaimableAmount` and the other being `getClaimedAmount`?